razorpay-payments.php

```diff
@@ -292,44 +324,40 @@ function check_razorpay_response()
 
                 $success = false;
                 $error = "";
+                $status = "";
 
                 $api = new Api($key_id, $key_secret);
-
+                
                 try
                 {
                     if ($this->payment_action === 'authorize')
                     {   
                         $payment = $api->payment->fetch($razorpay_payment_id);
             
-                        $expectedStatus = 'authorized';
+                        $status = 'authorized';
                     }
                     else
                     {
-                        $payment = $api->payment->fetch($razorpay_payment_id);
-                        $amount = $payment->amount;
-                        
-                        $payment = $payment->capture(array('amount'=>$amount)); 
+                        $razorpay_order_id = $woocommerce->session->get('razorpay_order_id');
+                        $razorpay_signature = $_REQUEST['razorpay_signature'];
                         
-                        $expectedStatus = 'captured';
+                        $signature = hash_hmac('sha256', $razorpay_order_id . '|' . $razorpay_payment_id, $key_secret);
+
+                        if( hash_equals($signature , $razorpay_signature) ){
+                            $status = 'captured';
```

Comment:
$status is never used as a string anywhere. Create a boolean `$captured` instead.